meta-mender-core/recipes-bsp/u-boot/patches/0002-Generic-boot-code-for-Mender.patch

```diff
@@ -58,6 +56,10 @@ index 0000000000..69a526805d
 +# error CONFIG_BOOTCOUNT_ENV is required for Mender to work. Make sure that: 1) All the instructions at docs.mender.io/devices/integrating-with-u-boot have been followed. 2) All required layers are included in bblayers.conf, including any board specific layers such as meta-mender-<board>
 +#endif
 +
++#ifndef CONFIG_SYS_REDUNDAND_ENVIRONMENT
++# error CONFIG_SYS_REDUNDAND_ENVIRONMENT is required for Mender to work. Make sure that: 1) All the instructions at docs.mender.io/devices/integrating-with-u-boot have been followed. 2) All required layers are included in bblayers.conf, including any board specific layers such as meta-mender-<board>
++#endif
```

Comment:
This would break the build on many U-Boot forks out there I am wondering if we can avoid this, but I do not see easy way of doing this unless we can detect U-Boot version in pre-processing.

Otherwise we probably need to keep two versions of this patch pre/post 2020.01

Reply:
Yes, I have the same concern. However, `CONFIG_SYS_REDUNDAND_ENVIRONMENT` was used even prior to them moving it into `Kconfig`, but then it was defined based on whether `CONFIG_ENV_OFFSET_REDUND` was defined. So I'm thinking one of two scenarios is going to happen:

1. The U-Boot build already defines it based on other conditions. If so this will just silently work.
2. The U-Boot build doesn't define it (at least not before getting here), which means the user has to do it. However, it has no ill effects as it either doesn't influence the build, or means the same thing as it does in v2020.01 and onwards.

Does that make sense?

It would be possible to conditionally detect this by looking inside the `Kconfig` files in the build, but given that defining `CONFIG_SYS_REDUNDAND_ENVIRONMENT` should have no ill effects, I'm leaning towards avoiding that additional complexity.

Reply:
> CONFIG_SYS_REDUNDAND_ENVIRONMENT was used even prior to them moving it into Kconfig

Ah, then it is not as a big issue as I thought and agree that we should avoid additional complexity.